### PR TITLE
Initial release

### PR DESCRIPTION
**Pull request by joakimhellum**
Initial release of Terraform provider for [Domeneshop API v0](https://api.domeneshop.no/docs/).

This pull request adds the following:
- Adds support for [basic authentication](https://api.domeneshop.no/docs/#section/Authentication/basicAuth)
- Adds support for the following data sources: `domeneshop_domain`, `domeneshop_domains`, `domeneshop_forward`, `domeneshop_forwards`, `domeneshop_record`, `domeneshop_records`
- Adds support for the following resources: `domeneshop_forward`, `domeneshop_record`
- Adds tests for data sources and resources
- Adds support for importing resources using `terraform import` (see [Resources should be importable](https://www.terraform.io/docs/extend/hashicorp-provider-design-principles.html#resources-should-be-importable))
- Adds provider docs generated by [hashicorp/terraform-plugin-docs](https://github.com/hashicorp/terraform-plugin-docs)

**Note 1:** The current API specification has multiple issues and is incomplete (see [API Documentation](https://api.domeneshop.no/docs/) and [OpenAPI specification](https://api.domeneshop.no/51bbc27b-bdb0-4a2d-8631-672d61eb98c0)), which makes full code generation (API client and Terraform provider) very complicated at the moment, so we use partial code generation. We will use full code generation once updated API specification is avilable from provider.

**Note 2:** Uses experimental version of [`innovationnorway/go-domeneshop`](https://github.com/innovationnorway/go-domeneshop) API client. A stable version will be released once updated API specification is available from the provider. 

**Note 3:** The Domeneshop API is in version 0 (see [Testing period](https://api.domeneshop.no/docs/#section/Testing-period)), which means that the API that could drift unexpectedly. Acceptance tests have been scheduled to run at regular interval to ensure we are alerted if the provider breaks due to API changes (See [Actions](https://github.com/innovationnorway/terraform-provider-domeneshop/actions?query=workflow%3ATests) tab).

### CODE REVIEW

**Comment by joakimhellum** [comment]
All [acceptance tests](https://github.com/innovationnorway/terraform-provider-domeneshop/runs/1677337992?check_suite_focus=true#step:5:1) were successful

image

**Review by roberthstrand — APPROVED** [review]
🤘

**Comment by joakimhellum** [comment]
Following changes have been made since @roberthstrand reviewed:
- Added HTTP forwards data sources and resource
- Added support for importing resources

All [acceptance tests](https://github.com/innovationnorway/terraform-provider-domeneshop/pull/3/checks?check_run_id=1682396128#step:5:1) were successful.

image

Will go ahead and merge this, then create `v0.1.0` release.

FYI @innovationnorway/platform

File: internal/provider/resource_record.go [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+			"domain_id": {
+				Description: "ID of the domain.",
+				Type:        schema.TypeInt,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"host": {
+				Description: "The host/subdomain the DNS record applies to.",
+				Type:        schema.TypeString,
+				Required:    true,
```
Comment (joakimhellum) [inline internal/provider/resource_record.go:36]:
```suggestion
				Required:    true,
				ForceNew:  true,
```

File: internal/provider/resource_record.go [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+			},
+			"type": {
+				Description: "The type of the record. Possible values are: `A`, `AAAA`, `ANAME`, `CNAME`, `DS`, `MX`, `NS`, `SRV`, `TXT`, `TLSA`.",
+				Type:        schema.TypeString,
+				Required:    true,
```
Comment (joakimhellum) [inline internal/provider/resource_record.go:42]:
```suggestion
				Required:    true,
				ForceNew:  true,
```

**Comment by joakimhellum** [comment]
All [acceptance tests](https://github.com/innovationnorway/terraform-provider-domeneshop/pull/3/checks?check_run_id=1683020972#step:5:48) were successful (again), will merge this 🥳

**Pull request merged**